common/src/main/java/org/opensearch/sql/common/utils/QueryContext.java

```diff
@@ -51,6 +55,31 @@ public static String getRequestId() {
     return id;
   }
 
+  /**
+   * Capture error message to aggregate diagnostics
+   * for both legacy and new SQL engines.
+   * Can be deleted once the
+   * legacy SQL engine is deprecated.
+   */
+  public static void setError(String error) {
```

Comment:
do we have better way to pass error messsage around instead of using threadcontext? we should consider deprecated threadcontext instead of add more capability to it.

Reply:
This feature (aggregating error messages from V1 and V2) will be removed soon along with V1. I can create/use a thread-safe singleton Map instead.

Reply:
Revered in b5ad3c9.